### DISCUSSION TITLE
Blender Triplanar Material Import

### DISCUSSION THREAD

**Discussion by natedawggh** [godotengine/godot-proposals#6342] [3D]
GODOT's new Blender import is fantastic. It doesn't automatically import triplanar materials, however, even though these are often used by developers, like myself. The blender image texture node contains a projection option ("flat," "box," "sphere," "tube") it seems like an easy feature to implement that GODOT's triplanar texture option would be set to "on" if the box projection for the image texture node is selected, since that is how triplanar is done in blender. As many game assets use triplanar materials this would make the material pipeline so much more efficient. Also, the relevant transform settings for UV1 scale and offset in these materials are not the default (meaning that blender is affecting them) but it's not clear where this change is coming from in blender, as my coordinate nodes are not the same as what shows up in GODOT on import, and would be great if it were either the default 1,1,1 or connected to some other obvious node or transform in blender.
<img width="247" alt="Screenshot 2023-02-22 at 3 21 51 PM" src="https://user-images.githubusercontent.com/125102099/220773324-3e000ff4-5bb2-45c7-b9bd-d82e9bb1eaa8.png">

**Comment by Calinou** [comment]
This is something that should be implemented in the glTF exporter/importer, as Godot doesn't import Blender files directly. Instead, Godot internally calls Blender's glTF export function, which is then read by Godot's glTF importer.

**Comment by iganinja** [comment]
Any news about this feature? An alternative would be to create an import script to enable triplanar flag, but it would be great to add this functionality to glTF importer.

**Reply by Calinou** [reply]
To my knowledge, nobody is currently working on implementing this. I think this is only really feasible if glTF has an extension for triplanar materials. Otherwise, there is no way Godot could read the fact that the material is supposed to use triplanar projection.
